### PR TITLE
feat: Add Keycloak module

### LINKED ISSUE #879: [Enhancement]: Add Keycloak module
### Problem

There is no support for Keycloak OIDC

### Solution

Add support for Keycloak container

### Benefit

We can run integration tests for services that used Keycloak authentication and authorization

### Alternatives

No alternatives

### Would you like to help contributing this enhancement?

No

### PR DESCRIPTION
**Pull request by VMelnalksnis**
## What does this PR do?

Introduce Keycloak module.

## Why is it important?

Allows to run integration tests with OAuth/OIDC auth.

## Related issues

Closes #879

### CODE REVIEW

File: src/Testcontainers.Keycloak/KeycloakBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+	/// <inheritdoc />
+	protected override KeycloakBuilder Init()
+	{
+		return base.Init()
+			.WithImage(KeycloakImage)
+			.WithCommand("start-dev")
+			.WithPortBinding(KeycloakPort, true)
+			.WithUsername(DefaultUsername)
+			.WithPassword(DefaultPassword);
```
Comment (HofmeisterAn) [inline src/Testcontainers.Keycloak/KeycloakBuilder.cs]:
```suggestion
		return base.Init()
			.WithImage(KeycloakImage)
			.WithCommand("start-dev")
			.WithCommand("--health-enabled", "true")
			.WithPortBinding(KeycloakPort, true)
			.WithUsername(DefaultUsername)
			.WithPassword(DefaultPassword)
			.WithWaitStrategy(Wait.ForUnixContainer().UntilHttpRequestIsSucceeded(request =>
				request.ForPath("/health/ready").ForPort(KeycloakPort)));
```

What about using Keycloak's [HEALTHCHECK](https://www.keycloak.org/server/health) to determine readiness? Then we can remove it in `WithUsername(string)`.

**Review by HofmeisterAn — COMMENTED** [review]
Thank you very much for your contribution. The Keycloak module configuration looks quite good. I just have a minor question or suggestion to improve it. Other than that, great work! Thank you 🥳.

File: src/Testcontainers.Keycloak/KeycloakBuilder.cs (hunk 1 above)
Comment (VMelnalksnis) [reply src/Testcontainers.Keycloak/KeycloakBuilder.cs]: Yeah, that looks better, forgot to check the documentation. Fixed and rebased latest changes.

**Review by HofmeisterAn — COMMENTED** [review]
PR looks good. Anything left? Ready for review?

**Comment by VMelnalksnis** [comment]
This should be it, unless I need to rebase the latest changes again.

**Comment by HofmeisterAn** [comment]
> This should be it, unless I need to rebase the latest changes again.

👍 No, you don't. I'll merge it later. Thanks already.

**Review by HofmeisterAn — APPROVED** [review]
Thanks for your contribution 🙌.

**Pull request merged**